### PR TITLE
Add gamecode fixes

### PR DESCRIPTION
**Pull request by xplshn**

### CODE REVIEW

**Comment by xplshn** [comment]
@AsciiWolf Any idea why this fails? I need to add the OAx folder to the flatpak, on my computer it works, if I manually add the folder, the launch script is not the problem here.

```
FB: Running: unzip /srv/buildbot/worker/build-x86_64-5/build/.flatpak-builder/downloads/5b92265b513a48df39bab7cb46479d8bf3582789400cbe3bd5c66ba0f8ef8b79/openarena_gamecode_2023-05-30_905b539.zip
Archive:  /srv/buildbot/worker/build-x86_64-5/build/.flatpak-builder/downloads/5b92265b513a48df39bab7cb46479d8bf3582789400cbe3bd5c66ba0f8ef8b79/openarena_gamecode_2023-05-30_905b539.zip
   creating: oax/
  inflating: oax/oax.pk3             
]2;flatpak-builder: Installing openarena-dataRunning: mv baseoa oax missionpack /app/share/openarena/
FB: Running 'flatpak build --die-with-parent --env=FLATPAK_BUILDER_BUILDDIR=/run/build/openarena-data --nofilesystem=host:reset --filesystem=/srv/buildbot/worker/build-x86_64-5/build/.flatpak-builder/build/openarena-data-1 --bind-mount=/run/build/openarena-data=/srv/buildbot/worker/build-x86_64-5/build/.flatpak-builder/build/openarena-data-1 --build-dir=/run/build/openarena-data --bind-mount=/run/ccache=/srv/buildbot/worker/build-x86_64-5/build/.flatpak-builder/ccache --unshare=network --env=SOURCE_DATE_EPOCH=1698446221 '--env=CFLAGS=-O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer ' '--env=CXXFLAGS=-O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer ' '--env=LDFLAGS=-L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed ' --env=CCACHE_DIR=/run/ccache/disabled --env=PATH=/app/bin:/usr/bin --env=LD_LIBRARY_PATH=/app/lib --env=PKG_CONFIG_PATH=/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig --env=FLATPAK_BUILDER_N_JOBS=64 /srv/buildbot/worker/build-x86_64-5/build/.flatpak-builder/rofiles/rofiles-CT79Fo /bin/sh -c 'mv baseoa oax missionpack /app/share/openarena/'' on host
mv: cannot stat 'oax': No such file or directory
FB: host_command_exited_cb 1285357 256
]2;flatpak-builder: Cleanup openarena-dataError: module openarena-data: Child process exited with code 1
FB: Unmounting read-only fs: fusermount -uz /srv/buildbot/worker/build-x86_64-5/build/.flatpak-builder/rofiles/rofiles-CT79Fo
program finished with exit code 1
elapsedTime=169.171631
```

It fails when it tries to move the oax folder, but the unzip command says it is there

**Comment by AsciiWolf** [comment]
What is OAx and why are you adding it? If it is some nightly, recent build of OpenArena, then it should go into the "beta" branch, definitely not "master".

**Comment by xplshn** [comment]
Oax contains gamecode fixes, and it won't be updated soon, yes it is the nightly version of OpenARENA(Not really, it doesn't contain anything apart from fixes to the 0.8.8 version).
For example, Oax fixes the video settings, it starts the game in your native resolution by default.
I tested playing on Ratmod servers, vanilla servers, OmegA servers... It could be a good inclusion to have this, maybe not by default, just there in the folder, so that if the user chooses to load it, using the CFG file, he will get these features or rather, fixes

**Comment by AsciiWolf** [comment]
Ah, makes sense. Anyway, I think that it needs additional testing before merging it into master. Any chance you could create a "beta" branch and merge it there? Thanks!

**Comment by xplshn** [comment]
Yeah, problem is, I don't know what is failing in the YAML, I need the flatpak to download the zip, unzip the OAX folder into the /app/share/openarena/ folder... i did use sha256 keys, and the URL works

**Comment by xplshn** [comment]
bot, build

**Comment by xplshn** [comment]
Its so weird that it failed when moving, not when unzipping and not even when verifying the files

**Comment by xplshn** [comment]
bot, build

**Comment by xplshn** [comment]
bot, build

File: io.github.ec_.Quake3e.OpenArena.yaml [hunk 1]
```diff
@@ -41,8 +41,11 @@ modules:
       - type: archive
         url: "https://downloads.sourceforge.net/project/oarena/openarena-0.8.8.zip"
         sha256: "5a8faf7f5b51f351b0a1618c06b6b98a5f1a6758f1d39818de2c87df2a0bac4a"
+      - type: archive
+        url: "https://files.poulsander.com/~poul19/public_files/oa/dev088/gamecode_nightly/openarena_gamecode_2023-05-30_905b539.zip"
+        sha256: "5b92265b513a48df39bab7cb46479d8bf3582789400cbe3bd5c66ba0f8ef8b79"
     build-commands:
-      - mv baseoa missionpack /app/share/openarena/
+      - mv baseoa oax missionpack /app/share/openarena/
```
Comment (AsciiWolf) [inline io.github.ec_.Quake3e.OpenArena.yaml:48]:
```suggestion
      - mv baseoa missionpack /app/share/openarena/
      - mkdir -p /app/share/openarena/oax && mv oax.pk3 /app/share/openarena/oax/
```

**Review by AsciiWolf — COMMENTED** [review]
I am not sure, but you may try this.

**Comment by xplshn** [comment]
Or do I pull to a "beta" branch?

**Comment by AsciiWolf** [comment]
It seems to work fine so feel free to merge it into master.

**Pull request merged**